src/platform/backends/shared/base_snapshot.h

```diff
@@ -72,13 +73,14 @@ class BaseSnapshot : public Snapshot
     {
     };
     BaseSnapshot(InnerJsonTag, const QJsonObject& json, const VirtualMachine& vm);
-    BaseSnapshot(const std::string& name, const std::string& comment, std::shared_ptr<const Snapshot> parent,
-                 int num_cores, MemorySize mem_size, MemorySize disk_space, VirtualMachine::State state,
-                 std::unordered_map<std::string, VMMount> mounts, QJsonObject metadata);
+    BaseSnapshot(const std::string& name, const std::string& comment, const QDateTime& creation_timestamp,
+                 std::shared_ptr<const Snapshot> parent, int num_cores, MemorySize mem_size, MemorySize disk_space,
+                 VirtualMachine::State state, std::unordered_map<std::string, VMMount> mounts, QJsonObject metadata);
 
 private:
     std::string name;
     std::string comment;
+    const QDateTime creation_timestamp;
```

Comment:
All other const fields are below, after a comment explaining them. Is there a reason why you need this order? Otherwise, could you move the timestamp there?